mergin/client.py

```diff
@@ -848,3 +848,18 @@ def resolve_unfinished_pull(self, directory):
         mp = MerginProject(directory)
         conflicts = mp.resolve_unfinished_pull(self.username())
         return conflicts
+
+    def have_writing_permissions(self, directory):
+        """
+        Test whether user has writing permissions on the given project.
+        We rely on the "permissions" field here, as it provides more accurate
+        information and take into account namespace permissions too.
+
+        :param directory: project's directory
+        :type directory: str
+        :returns: whether user has writing (upload) permission on the project
+        :rtype: bool
+        """
+        mp = MerginProject(directory)
+        info = self.project_info(mp.metadata["name"])
```

Comment:
is not it better to fetch it from server rather than rely on some cached information?